Activity log: Item data and generic design

sirreal opened the pull request:
This is not complete as some cleanup is required, but the concept is ready for some feedback.

I largely refactored `ActivityLogItem` and removed an unused chunk of logic from `ActivityLog` that was used mostly for mocking activity in _the big PR_.

I tried to capture `log` in `propTypes`, largely just to document what sorts of data are present.

I added some specific `icon` and `status` just to see what that would look like in code and on the screen.

### To test:
* https://calypso.live/stats/activity?branch=update/activity-log-item-genericstyles
* This is just a generic UI. Try to have a look at how different activity behave.
* Keep an eye on the console for warnings/errors.

### Screenshot:

<!-- old: ![activity](https://user-images.githubusercontent.com/841763/27478655-cf116b80-5810-11e7-9f29-ee8635a75306.png) -->

![al](https://user-images.githubusercontent.com/841763/27582628-dde81b9e-5b31-11e7-8f14-f8d0b9540594.png)


### Discussion points:

1. Raw `log` prop is passed as an object. How do we feel about that?
   * Easy for now as data may change
   * More difficult to work with props
   * More difficult to diff props (performance/render)
1. Should something like `fromApi` be normalizing all of the activity data?
1. Sub-components: ActivityLogItem is pretty big and a few subcomponents would work nicely. It would also allow flattening and passing a few selected props from `log`.
1. Actor Gravatar  - the actor does not include avatar_URL which is what the `Gravatar` component requires. Some possible solutions:
   * API includes URL
   * Fetch wpcom user if wpcom_user_id is included
1. Using some `switch`es in a few places. It feels good so far, allows for nice cascading. Another option to consider is using a big object that maps `name`s to `{ status, icon, ... }`.
1. We've essentially punted on `description` and the translation parts of this component after conversation in #15344 and p1498130501662286-slack-activity-log. The question remains of how to best deal with it.

### To do:
- [x] cleanup dev output
- [x] actor Gravatar solution

### Follow up work
* Restore and other action analytics
* Link activity to content? (post, comment...)
* Ellipsis menu actions - there's a lot of interesting actions that could happen here, like approving a comment right from the list. OTOH
* Add/update `status` and `icon` for different activity.
* break down into logical sub-components. Proposed sub-components (please discuss):
   * icon
   * actor
   * description
   * menu

sirreal commented:
> Actor Gravatar - the actor does not include avatar_URL which is what the Gravatar component requires. Some possible solutions:
> * API includes URL

@roccotripaldi Is this even possible/feasible?

seear commented:
Quick thoughts on these:

> Raw log prop is passed as an object. How do we feel about that?
> Easy for now as data may change
> More difficult to work with props
> More difficult to diff props (performance/render)

At this stage, I think it is fine as it is. I feel that the data we are getting from the endpoint could change shape a fair bit as we figure out exactly what we need, so we don't need to harden the design too much right now.

> Should something like fromApi be normalizing all of the activity data?

Again, it might be prudent to put this off until we are confident in the fields that we want to use.

roccotripaldi commented:
@sirreal - re: gravatar url

I think we should add the gravatar URL on the API. I will look into this.

sirreal commented:
Gravatar patch from @roccotripaldi D6143-code 🎉

sirreal commented:
I added Gravatars from D6143-code in 411ff4c 💥 

@keoshi The styleguide is great, it's too bad I can't ❤️  🎉  reviews.

I just pushed some changes to address the issues you mention, design should be pretty good at this point. In the near future we'll need to address the spacing in the event card (avatar, author, event title), there's some funny overlapping happening for me:

![overlap](https://user-images.githubusercontent.com/841763/27582417-182b3008-5b31-11e7-9537-20536ad0db09.png)

That said, I've updated the description to reflect follow up work. I think this PR addresses the main issues and gets the right data in the right places. I'd like to merge soon and continue with follow-up in subsequent PRs.

@seear @dereksmart Your feelings?

keoshi commented:
That is looking mighty good, @sirreal — thanks for working on that!

Re: overlapping: we should apply some flexbox magic there. I can help, once the support rotation is complete.

seear commented on client/my-sites/stats/activity-log-item/index.jsx:
```diff
@@ -2,115 +2,250 @@
-		status: 'is-info',
-		icon: 'info-outline'
+		log: PropTypes.shape( {
+			group: PropTypes.oneOf( [
```
`'theme'` is another valid value for `group`.
